tests/test-sources/plugins/lazy-load.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lazy-load-neovim-plugin =
+    { config, ... }:
+    {
+      plugins = {
+        lz-n = {
+          enable = true;
+        };
+
+        neotest = {
+          enable = true;
+          lazyLoad = {
+            enable = true;
```

Comment:
https://github.com/nix-community/nixvim/pull/2577#issuecomment-2509783543

Reply:
Global toggle and/or on by default should not be part of the initial implementation.

Down the line, I could envision the per-plugin options _defaulting_ to the value of some global option.

But the initial implementation should be as simple to implement as possible

Reply:
I think it should be by default when you have lazy loading through `lz-ln` or another one, enabled because It will just make things cleaner in codebases.

Reply:
I'm not saying we shouldn't do this. I'm saying we can have that debate and come up with a solution _after_ this PR.

This PR should be the minimal viable product, so to speak.